cpmpy/solvers/z3.py

```diff
@@ -249,10 +250,7 @@ def transform(self, cpm_expr):
         # Z3 supports nested expressions, so no transformations needed
         # that also means we don't need to extract user variables here
         # we store them directly in `solver_var()` itself.
-        if is_any_list(cpm_expr):
-            return cpm_expr
-        else:
-            return [cpm_expr]
+        return make_cpm_expr(cpm_expr)
 
     def _post_constraint(self, cpm_expr):
         """
```

Comment:
the if is_any_list check in post_constraint is not needed anymore because of the make_cpm_expr call in transform, right?